### PR TITLE
Xenomorph Acid Well

### PR DESCRIPTION
**Pull request by jjpark-kb**
## About The Pull Request
adds a new action and structure for xenos: the acid well.
the acid well is a structure (effect, similar to hugger traps) that can put out fires on xenos. It can store a total of five charges, each which cost 200 plasma. There can ONLY be five wells as well in total, so make sure to destroy your wells if you leave the planet.
Be warned marines, do not step on the acid wells, because they WILL hurt you
Be warned Queens, you are no longer fire resistant now, those wells are for you.
Wells can only be placed by queens and shrikes.

Credits to Kuro for the sprites.

## Why It's Good For The Game
adds more structures for xenos to play around with. With the addition of adding a "water tile", I was able to remove the fire resistance that queens have. This will also open up opportunities for new kinds of maps without the need to worry about water etc.

## Changelog
:cl:
add: Added acid wells
del: Removed queen's fire resistance
/:cl:

### CODE REVIEW

**Comment by Glcbitj** [comment]
Kinda cool idea but you should allow for Drones and Hivelords to build such structures, you know, that's kinda their thing.

**Comment by Max-023** [comment]
This is an odd way to try to nerf the queen. Fixed xeno hold points have never been a thing with xenos due to their high mobility playstyle so I don't really see this as being useful and more of just a fuck you to the queen, who is already one of the weakest castes.

**Comment by terranaut1** [comment]
xenos rely on hit and run and robbing queen of her fire immunity removes her utility as a pusher, it returns to a state of having to avoid fire and dealing with it when it happens instead of being able to ignore it, you just gave more options to deal with it. except those options dont fit xeno playstyle at all.

File: code/game/objects/structures/xeno.dm [hunk 1]
```diff
@@ -695,3 +695,121 @@ TUNNEL
...
+/obj/effect/alien/resin/acidwell/Crossed(atom/A)
+	. = ..()
+	if(iscarbon(A))
+		HasProximity(A)
+
+/obj/effect/alien/resin/acidwell/HasProximity(atom/movable/AM)
+	if(!iscarbon(AM))
+		return
+	if(!(AM in GLOB.alive_living_list))
+		return
```
Comment (psykzz) [inline code/game/objects/structures/xeno.dm:796]: You gotta stop doing this `in` checks they are not good.
If your looking to check if the mob is alive, just type caste it and check `stat == DEAD`

**Comment by davidchan13** [comment]
I'm not fond of this change but I'm overall neutral on the idea as I've been saying for months we need more varied resins structures.

But, it's a complete :-1: from me if we are keeping structures out of Drone/Hivelord hands, it's their primary job to craft the defences and support the rest of the hive. Let's not give them an imcomplete toolbox, please.

File: code/game/objects/structures/xeno.dm (hunk 1 above)
Comment (jjpark-kb) [reply code/game/objects/structures/xeno.dm]: I believe I fixed this. Sorry about that.

**Comment by jjpark-kb** [comment]
> I'm not fond of this change but I'm overall neutral on the idea as I've been saying for months we need more varied resins structures.
> 
> But, it's a complete 👎 from me if we are keeping structures out of Drone/Hivelord hands, it's their primary job to craft the defences and support the rest of the hive. Let's not give them an imcomplete toolbox, please.

Hivelords already have it, I was skeptical of giving it to drones just because they already have healing. Paired with an ability to put out fires now as well as healing felt extremely strong.

**Comment by psykzz** [comment]
Having 2nd thoughts on drone as a Tier 1.

If you want the better structures, like acid wells you should be forced to tier up to hivelord.
Otherwise there is gonna be no reason to evolve from drone.

**Comment by jjpark-kb** [comment]
> Having 2nd thoughts on drone as a Tier 1 have these abilities.
> 
> If you want the better structures, like acid wells you should be forced to tier up to hivelord.
> Otherwise there is gonna be no reason to evolve from drone.

Will do.

**Pull request merged**